File: 258-add-digits/258-add-digits.py
class Solution:
    def addDigits(self, num: int) -> int:
        if num < 10:
            return num
        result = 0
        while num != 0:
            result = result + (num % 10)
            num = int(num / 10)
        return self.addDigits(result)

File: 258-add-digits/258-add-digits.kt
class Solution {
    fun addDigits(num: Int): Int {
        val list: MutableList<Int> = emptyList<Int>().toMutableList()
        var number = num

        while(number >= 10){
            while(number >= 10){
                list.add(number / 10)
                number %= 10
            }
            list.add(number)
            number = list.sum()
            list.clear()
        }

        return number
    }
}